lib/run.js

```diff
@@ -93,13 +94,25 @@ function _run (opts, cb, tracker) {
   let numRunning = opts.connections
   let startTime = Date.now()
   const includeErrorStats = !opts.excludeErrorStats
+  let form
+
+  if (opts.form) {
+    try {
+      form = multipart(opts.form)
+    } catch (error) {
+      errorCb(error)
+      return
+    }
+  }
 
   // copy over fields so that the client
   // performs the right HTTP requests
   url.pipelining = opts.pipelining
-  url.method = opts.method
-  url.body = opts.body
-  url.headers = opts.headers
+  url.method = opts.form ? 'POST' : opts.method
```

Comment:
I think we should honor the method option even if there is a form

Reply:
I have restored
```js
url.method = opts.method
```

but I am not sure that is what you meant anymore.

Maybe I could have assigned a default and let the user override it?

```js
if (opts.form) {
  opts.method = opts.method || 'POST'
}
```

Reply:
Yes, exactly.